src/Lib/Controller.php

```diff
@@ -204,6 +204,46 @@ protected function draw($view = null, $dir = null)
             ->fetch($view, $dir);
     }
 
+    /**
+     * Overwritable method called before the partial method
+     *
+     * @param string $view The name of the view file to render
+     * @param string $dir The directory to find the given view in
+     * @return array A list containing: (optional)
+     *
+     *  - view The name of the view file to render
+     *  - params An array of parameters to set in the view
+     *  - dir The directory to find the given view in
+     */
+    public function prePartial($view, $params = [], $dir = null)
```

Comment:
I think the `$params = null` you had first would suffice, consistent with `::partial`, or perhaps `array $params = null` to enforce the type. The argument should also be documented for `::prePartial` and `::postPartial`.